services/mac_manager/src/main/java/com/futurewei/alcor/macmanager/dao/MacPoolRepository.java

```diff
@@ -0,0 +1,126 @@
+/*Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.macmanager.dao;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.repo.ICacheRepository;
+import com.futurewei.alcor.macmanager.entity.MacAddress;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Vector;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Repository
+@ComponentScan(value = "com.futurewei.alcor.common.db")
+public class MacPoolRepository implements ICacheRepository<MacAddress> {
+    private ICache<String, MacAddress> cache;
+
+    @Autowired
+    public MacPoolRepository(CacheFactory cacheFactory) {
+        cache = cacheFactory.getCache(MacAddress.class);
+    }
+
+    public ICache<String, MacAddress> getCache() {
+        return cache;
+    }
+
+    @PostConstruct
+    private void init() {
+
+    }
+
+    @Override
+    public MacAddress findItem(String strMacAddress) throws CacheException, ResourceNotFoundException {
+        if (cache.containsKey(strMacAddress))
+            return cache.get(strMacAddress);
+        else
+            return null;
+    }
+
+    @Override
+    public Map<String, MacAddress> findAllItems() throws CacheException {
+        return cache.getAll();
+    }
+
+    /**
```

Comment:
Good to see Comments here. Let us try to do the same for important methods throughout NodeManager, including but not limited to WebLayer and ServiceLayer.

Reply:
Yes, I agree.